### PR TITLE
CVE-2013-2842 and CVE-2011-1186

### PR DESCRIPTION
**Pull request by paulhulbert**

### CODE REVIEW

File: cves/CVE-2011-1186.yml [hunk 1]
```diff
@@ -170,4 +170,4 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: This one was tough to call a mistake because it was highly unlikely to occur except during debug.  Printing almost never happens concurrently, and it was only occasionally reproducable anyway.  The mistake was assuming that the printing was always thread safe.  They chose not to fix the issue because it would have required more work than it was worth.  Instead, they just fixed the particular issue so that it would go back to looking like it worked.
```
Comment (jmiller656) [inline cves/CVE-2011-1186.yml]: This line was a bit long. Content was fine, just would be nice to add the occasional line break for long text like this

File: cves/CVE-2013-2842.yml [hunk 2]
```diff
@@ -37,11 +38,9 @@ fixes_vcc_instructions: |
...
   CVE-2011-3092.yml). Fixes and VCCs follow the same format.
 fixes:
-- :commit: 25077160626d9cad4d4d658d67482c465d165b20
-  :note: ''
-- :commit: 25077160626d9cad4d4d658d67482c465d165b20
-  :note: ''
-vccs: []
+- :commit: 5f6d1958c7a1c7c5dcd7c12d7cbf2043b046979f
+vccs: 
+- :commit: 199fc2603c022f2ad7bc19b4a6b16136808b37f1
```
Comment (jmiller656) [inline cves/CVE-2013-2842.yml]: I'm pretty sure the vccs line can just be 

-vccs: [199fc2603c022f2ad7bc19b4a6b16136808b37f1]

Also, it looks like you changed what the fix commits were. Just double checking to make sure that's what you intended.

File: cves/CVE-2013-2842.yml [hunk 3]
```diff
@@ -16,6 +16,7 @@ description_instructions: |
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
 description: 
+  When a frame was removed from its parent frame, it could still be referenced.  This could lead to a use-after-free error.  It could cause a DOS or potentially give an attacker the ability to interact with the user's computer.
```
Comment (jmiller656) [inline cves/CVE-2013-2842.yml]: Kinda curious about what you mean by 'interact'. As in arbitrary code execution? You might want to be more specific if that's the case.

File: cves/CVE-2011-1186.yml [hunk 4]
```diff
@@ -92,10 +94,8 @@ interesting_commits:
     interesting in light of the lessons learned from this vulnerability. Any
     emerging themes?
   commits:
-  - commit: 
-    note: 
-  - commit: 
-    note: 
+  - commit: 8e2b6472071f38c065a3d00adb136ef259ef68a1
+    note: Created a resource message filter
```
Comment (hinza0188) [inline cves/CVE-2011-1186.yml]: 1 up vote

File: cves/CVE-2011-1186.yml [hunk 5]
```diff
@@ -156,8 +156,8 @@ lessons:
     applies: 
     note: 
   complex_inputs:
-    applies: 
-    note: 
+    applies: true
+    note: This is the best one to describe it.  It was not a complex input, so much as it was a chance of getting two inputs at once, which would cause it to crash.
```
Comment (hinza0188) [inline cves/CVE-2011-1186.yml]: I guess you could re-phrase this as "Race condition input is complex enough for the chrome...?"
1 up vote

File: cves/CVE-2011-1186.yml (hunk 1 above)
Comment (hinza0188) [inline cves/CVE-2011-1186.yml]: 1 up vote

File: cves/CVE-2013-2842.yml (hunk 3 above)
Comment (hinza0188) [inline cves/CVE-2013-2842.yml]: Agreed, @jmiller656

File: cves/CVE-2013-2842.yml (hunk 2 above)
Comment (hinza0188) [inline cves/CVE-2013-2842.yml]: The auto-found two fix commits had exactly same hash. 

As @jmiller656 mentioned, I'd check if the removed commits are really garbage commit. It seems like 

saving both removed fix commit (25077160626d9cad4d4d658d67482c465d165b20)

and the newly found fix commit (5f6d1958c7a1c7c5dcd7c12d7cbf2043b046979f)

in the .yml file

File: cves/CVE-2013-2842.yml [hunk 6]
```diff
@@ -63,8 +62,8 @@ unit_tested:
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
   answer:
```
Comment (hinza0188) [inline cves/CVE-2013-2842.yml]: you need the written answer for this unit tested section

File: cves/CVE-2013-2842.yml [hunk 7]
```diff
@@ -93,7 +92,7 @@ subsystem:
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged.
   answer:
```
Comment (hinza0188) [inline cves/CVE-2013-2842.yml]: you need written answer of the description of blink/widgets subsystem

File: cves/CVE-2013-2842.yml [hunk 8]
```diff
@@ -102,10 +101,14 @@ interesting_commits:
     interesting in light of the lessons learned from this vulnerability. Any
     emerging themes?
   commits:
-  - commit: 
-    note: 
-  - commit: 
-    note: 
+  - commit: 4519e250d4187da2e5ebbc1f5900b9649f70b0d6
+    note: Removed the code to support Objective C
```
Comment (hinza0188) [inline cves/CVE-2013-2842.yml]: why removing objective C support interesting commit for this issue?

File: cves/CVE-2013-2842.yml [hunk 9]
```diff
@@ -102,10 +101,14 @@ interesting_commits:
...
+  - commit: bc34bb1203abd435dea2f0a62bbd05f9eea24656
+    note: Made a bunch of paths absolute in this commit and the next few commits.
+  - commit: 905b0461e247bd3f404d2bc437e43a82841823e0
+    note: The first attempt to fix this bug
```
Comment (hinza0188) [inline cves/CVE-2013-2842.yml]: 1 up vote, but could you describe what the attempt was?

File: cves/CVE-2013-2842.yml [hunk 10]
```diff
@@ -148,8 +151,8 @@ lessons:
     applies: 
     note: 
   distrust_input:
-    applies: 
-    note: 
+    applies: true
+    note: While there was not really input to the system, it could be attacked by closing certain parts of it at the wrong time.  The fix was to check to make sure that those parts were not closed.
```
Comment (hinza0188) [inline cves/CVE-2013-2842.yml]: 2 upvote

File: cves/CVE-2013-2842.yml [hunk 11]
```diff
@@ -180,4 +183,4 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: The mistake was that they freed a portion of memory that went along with a frame in the renderer, but they did not remove all references to it.  The fix was to check the frame before it was used to make sure that it had not been freed yet.
```
Comment (hinza0188) [inline cves/CVE-2013-2842.yml]: 3 up vote

File: cves/CVE-2013-2842.yml (hunk 2 above)
Comment (paulhulbert) [inline cves/CVE-2013-2842.yml]: @jmiller656 Meneely said that the default was the bracketed VCCs, but that we should use the same format as the fix commits.

Comment (paulhulbert) [inline cves/CVE-2013-2842.yml]: Yes, I intentionally switched the VCC commit hash.  The auto-found one was an attempted fix to the bug that failed.  The VCC was earlier.

File: cves/CVE-2011-1186.yml [hunk 12]
```diff
@@ -53,8 +55,8 @@ unit_tested:
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
   answer:
```
Comment (paulhulbert) [inline cves/CVE-2011-1186.yml]: Correct.  It was deemed too low a risk/harm to actually be worth testing.

File: cves/CVE-2011-1186.yml [hunk 13]
```diff
@@ -70,10 +72,10 @@ discovered:
 
     If there is no evidence as to how this vulnerability was found, then you may
     leave this part blank.
-  answer: 
-  date: 
-  automated: 
-  google: 
+  answer: Found when trying to fix issue 64051, a bug that is still marked as confidential.
```
Comment (paulhulbert) [inline cves/CVE-2011-1186.yml]: I cannot view 64051.  It is still confidential.

File: cves/CVE-2011-1186.yml [hunk 14]
```diff
@@ -70,10 +72,10 @@ discovered:
...
+  date: Dec 15 2010
+  automated: false
+  google: false
   contest:
```
Comment (paulhulbert) [inline cves/CVE-2011-1186.yml]: It was found by a random person working on a different bug.  Not by a chromium member.

File: cves/CVE-2011-1186.yml [hunk 15]
```diff
@@ -30,8 +30,10 @@ fixes_vcc_instructions: |
   CVE-2011-3092.yml). Fixes and VCCs follow the same format.
 fixes:
 - :commit: 032682b529bf96d184f409fb39a95f9c0d875f76
-  :note: ''
-vccs: []
+  :note: 'They did not actually fix the root causes, they just made it pass tests because it was not a security issue, and it was almost impossible for it to cause problems.'
+vccs: 
+- :commit: ccd0226c79553e318657d6285c2feacebd105996
+  :note: 'This commit added a safety check, but it did not check for a race condition'
```
Comment (paulhulbert) [inline cves/CVE-2011-1186.yml]: @hinza0188 Could you put all your upvotes in one comment?  I can't really tell if you're upvoting a comment in some places, or upvoting this CWE.

File: cves/CVE-2011-1186.yml [hunk 16]
```diff
@@ -83,7 +85,7 @@ subsystem:
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged.
   answer:
```
Comment (paulhulbert) [inline cves/CVE-2011-1186.yml]: Clarified

File: cves/CVE-2011-1186.yml (hunk 1 above)
Comment (paulhulbert) [inline cves/CVE-2011-1186.yml]: Done.

File: cves/CVE-2011-1186.yml [hunk 17]
```diff
@@ -15,7 +15,7 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: Chrome on Linux had a bug where if two calls to the print method were made in parallel, it could crash.  This was a potential DOS flaw.
```
Comment (paulhulbert) [inline cves/CVE-2011-1186.yml]: Clarified.

File: cves/CVE-2011-1186.yml (hunk 5 above)
Comment (paulhulbert) [inline cves/CVE-2011-1186.yml]: Removed this because after further investigation, I decided it didn't fit.

File: cves/CVE-2013-2842.yml (hunk 8 above)
Comment (paulhulbert) [inline cves/CVE-2013-2842.yml]: It was the only large commit in between the VCC and the fix.  Everything else was just minor refactoring.

File: cves/CVE-2013-2842.yml (hunk 9 above)
Comment (paulhulbert) [inline cves/CVE-2013-2842.yml]: Not without using very Chrome specific words and function names.  It was just changing the order of some function calls.

File: cves/CVE-2013-2842.yml (hunk 3 above)
Comment (paulhulbert) [inline cves/CVE-2013-2842.yml]: Reworded.

File: cves/CVE-2013-2842.yml (hunk 6 above)
Comment (paulhulbert) [inline cves/CVE-2013-2842.yml]: Fixed.

File: cves/CVE-2013-2842.yml (hunk 7 above)
Comment (paulhulbert) [inline cves/CVE-2013-2842.yml]: Fixed.

File: cves/CVE-2013-2842.yml (hunk 9 above)
Comment (hinza0188) [inline cves/CVE-2013-2842.yml]: Got it

File: cves/CVE-2011-1186.yml (hunk 15 above)
Comment (hinza0188) [inline cves/CVE-2011-1186.yml]: @paulhulbert gottya. I will put total score up vote at the end of the comment for each file

File: cves/CVE-2011-1186.yml [hunk 18]
```diff
@@ -170,4 +172,7 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
```
Comment (hinza0188) [inline cves/CVE-2011-1186.yml]: 8 up vote

File: cves/CVE-2013-2842.yml [hunk 19]
```diff
@@ -180,4 +185,6 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    The mistake was that they freed a portion of memory that went along with a frame in the renderer, but they did not remove all references to it.  
+    The fix was to check the frame before it was used to make sure that it had not been freed yet.
```
Comment (hinza0188) [inline cves/CVE-2013-2842.yml]: 6 up vote

File: cves/CVE-2011-1186.yml (hunk 15 above)
Comment (paulhulbert) [inline cves/CVE-2011-1186.yml]: @hinza0188 Awesome.  Thanks.

File: cves/CVE-2011-1186.yml (hunk 18 above)
Comment (jmiller656) [inline cves/CVE-2011-1186.yml]: 2 upvotes

File: cves/CVE-2013-2842.yml (hunk 19 above)
Comment (jmiller656) [inline cves/CVE-2013-2842.yml]: 3 upvotes

**Comment by hinza0188** [comment]
Thought upvote limit was 10 for each file while it was total of 10 for everything. gonna have to revise up vote number :( sry

**Pull request merged**